### DISCUSSION TITLE
Do group AssetSelections passed to define_asset_job take dependencies into consideration?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18221] [Q&A] [answered]
Simple question to double-check before I rework something unwieldy...

If I set up a job using `define_asset_job` and include `selection=AssetSelection.groups(GRP01, GRP02, GRP03, ...)` - will Dagster configure this job in a way that respects the asset dependencies across groups? For example, if assets in `GRP03` have dependencies to assets in `GRP01`, those will be taken into consideration when the job executes?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700658845633129?thread_ts=1700658845.633129&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
Yes - asset dependencies will be properly respected among all selected assets in a job, regardless of which `AssetSelection` type is used.
